README.md

```diff
@@ -198,3 +198,84 @@ func Function(
+
+#### Configuration
+
+Config files for the sync tool live in the [internal/sync/configs](./internal/sync/configs/) directory. A good example of all the features available is in the [nvidia-config.yaml](./internal/sync/configs/nvidia-config.yaml) file.
+
+Each config is a YAML document with a top level list of objects that follow this form:
+
+```yaml
+- name: operator-repo # 1
+  sync: true # 2
+  repo_link: "https://github.com/operator/repo" # 3
+  branch: main # 4
+  remote_api_directory: pkg/apis/v1 # 5
+  local_api_directory: schemes/operator/operatortypes # 6
+  replace_imports: # 7
+    - old: '"github.com/operator/repo/pkg/apis/v1/config"' # 8
+      new: '"github.com/openshift-kni/eco-goinfra/pkg/schemes/operator/operatortypes/config"'
+    - old: '"github.com/operator/repo/pkg/apis/v1/utils/exec"'
+      new: exec "github.com/openshift-kni/eco-goinfra/pkg/schemes/operator/operatortypes/executils" # 9
+  excludes:
+    - "*_test.go" # 10
+```
+
+1. Name, which does not have to be unique, identifies the repos in logs and controls where it is cloned during the sync. It does not affect the outcome of the sync itself and can be set as desired.
+2. Sync is whether the operator repo will be synced both periodically and when `make lib-sync` is used manually.
+3. Repo link is the url of the operator repo itself. It does not need to end in `.git`.
+4. Branch is the branch of the operator repo to sync with.
+5. Remote API directory is the path in the operator repo to sync with, relative to the operator repo root.
+6. Local API directory is the path in eco-goinfra where the remote API directory should be synced to. Relative to the eco-goinfra root, it should start with `schemes/`.
+7. Since the operator may import other parts of the repo or other repos in the same organization, this will need to be replaced with imports from eco-goinfra once synced.
+8. Import replacement is done through find and replace, so we include the double quotes to make sure it is an import being matched.
+9. If the package name in eco-goinfra is different than the operator repo, the import should be renamed so the code still works.
+10. Excludes is a list of file patterns to exclude. Since tests and mocks may add their own dependencies, excluding them can keep the amount of code that needs synced much smaller
+
+Like in the [nvidia-config.yaml](./internal/sync/configs/nvidia-config.yaml) example, it is often the case that one repo will import a few others. All of the imported repos should be specified in the sync config to avoid adding new dependencies.
+
+#### Using the Synced Types
+
+Instead of directly adding the scheme synced from the operator repo to the [clients](./pkg/clients/) package, the client scheme should be updated in `NewBuilder()` or `Pull()` functions. The [metallb](./pkg/metallb/metallb.go) package provides a good example of this.
+
```

Comment:
It'll also allow use to make packages more modular and reduce needed dependencies during the package import.